projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -703,7 +703,8 @@ export class IgxOverlayService implements OnDestroy {
         }
 
         // set content div width only if element to show has width
-        if (info.initialSize.width !== 0) {
+        // excluding the esf filtering as its initial width is the same as grid's width and is resized after the element is moved to the overlay
+        if (info.initialSize.width !== 0  && info.elementRef.nativeElement.tagName?.toLowerCase() !== 'igx-grid-excel-style-filtering') {
```

Comment:
Please no. Overlay has no knowledge about any component and it should not have!
I see two ways to fix this:
- easy one. Handle `contentAppended` event of the overlay. Get the overlay info and set its `initialSize` to (0; 0).
- hard one. We can remove `initialSize` from overlay info. This will require removing of second parameter of `IPositionStrategy.position` method. Also fixing of drop down as, if I am not missing something, this is the only component (and all the ones inheriting it) that benefits from setting the `width` of the overlay content element.